### PR TITLE
feat: formatting checks in CI

### PR DESCRIPTION
**Pull request by hamirmahal**

### CODE REVIEW

**Review by hamirmahal — COMMENTED** [review]
Pull request checks might have to be enabled in the repository settings.

**Review by lpil — COMMENTED** [review]
Nice! Can you also add running the compiler to this workflow too? May as well check that too. Thanks

**Comment by hamirmahal** [comment]
You're welcome!

**Review by lpil — COMMENTED** [review]
Could you put them in the same workflow please. No reason to waste lots of resources and extra carbon emissions by duplicating all the setup work. Thanks!

**Comment by hamirmahal** [comment]
You're welcome!

**Review by hamirmahal — COMMENTED** [review]
Okay, `test-compile` is now in the same workflow at the formatting check.

Let me know if there are any other changes you'd like me to make here.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - run: npm ci
+      - run: npm run format-check
+  test-compile:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - run: npm ci
+      - run: npm run test-compile
```
Comment (lpil) [inline .github/workflows/ci.yml:15]: This is still duplicating the same initialisation twice on two runners, let's remove that wasted work to make it faster and to consume less compute
```suggestion
  test:
    runs-on: ubuntu-latest
    steps:
      - run: npm ci
      - run: npm run test-compile
      - run: npm run format-check
```

**Review by lpil — APPROVED** [review]
Thank you!!

**Comment by lpil** [comment]
Hello! The build failed. Could you fix it please 🙏

Thank you.

**Comment by hamirmahal** [comment]
You're welcome!

**Comment by hamirmahal** [comment]
I'm taking a look right now.

**Pull request merged**